### PR TITLE
add multus network documentation

### PR DESCRIPTION
**Pull request by phoracek**

### CODE REVIEW

**Comment by phoracek** [comment]
@fabiand @yuvalif @SchSeba could you please take a look?

**Comment by phoracek** [comment]
[[epic-L2-connectivity] add multus documentation to kubevirt](https://trello.com/c/rji50OCU/197-epic-l2-connectivity-add-multus-documentation-to-kubevirt)

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 1]
```diff
@@ -18,34 +18,83 @@ All possible configuration options are available in the
...
+          bridge: {}
+  networks:
+  - name: default
+    pod: {} # Stock pod network
+```
+
+### multus
+
+It is also possible to connect VMI to a secondary networks using
+[Multus](https://github.com/intel/multus-cni). This assumes multus plugin is
```
Comment (rmohr) [inline workloads/virtual-machines/interfaces-and-networks.md]: This assumes that multus plugin**s** are installed

or 

This assumes that **a/the** multus plugin is installed

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 2]
```diff
@@ -18,34 +18,83 @@ All possible configuration options are available in the
...
+spec:
+  config: '{
+      "cniVersion": "0.3.1",
+      "type": "ovs",
+      "bridge": "br1",
+      "vlan": 100
+    }'
+```
+
+With following definition VMI will be connected to default pod network and also
```
Comment (rmohr) [inline workloads/virtual-machines/interfaces-and-networks.md]: , the VMI will be connected the default pod network and to the secondary Open vSwitch network.

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 3]
```diff
@@ -18,34 +18,83 @@ All possible configuration options are available in the
...
+  networks:
+  - name: default
+    pod: {} # Stock pod network
+```
+
+### multus
+
+It is also possible to connect VMI to a secondary networks using
+[Multus](https://github.com/intel/multus-cni). This assumes multus plugin is
+installed accross your cluster and NetworkAttachmentDefinition CRD was created.
```
Comment (rmohr) [inline workloads/virtual-machines/interfaces-and-networks.md]: and **a corresponding** `NetworkAttachmentDefinition` CRD exists.

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 4]
```diff
@@ -18,34 +18,83 @@ All possible configuration options are available in the
...
+
+First of all, NetworkAttachmentDefinition for desired secondary network must be
```
Comment (rmohr) [inline workloads/virtual-machines/interfaces-and-networks.md]: First, **a** `NetwokrAttachmentDefinition` for **the** desired secondary network must be created.

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 5]
```diff
@@ -18,34 +18,83 @@ All possible configuration options are available in the
...
+created. Following example defines a network implemented by [ovs-cni
```
Comment (rmohr) [inline workloads/virtual-machines/interfaces-and-networks.md]: **The** following example defines a network which uses  the [ovs-cni-plugin](...), which will connect **the** VMI

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 6]
```diff
@@ -18,34 +18,83 @@ All possible configuration options are available in the
...
+plugin](https://github.com/kubevirt/ovs-cni) that will connect VMI to Open
+vSwitch bridge `br1` and VLAN 100. Other CNI plugins such as ptp, bridge,
+macvlan or flannel might be used as well, for their installation and usage refer
+to respective project documentation.
+
+```yaml
```
Comment (rmohr) [inline workloads/virtual-machines/interfaces-and-networks.md]: You may want to add that his is normally pre-defined by the administrator.

**Comment by phoracek** [comment]
@rmohr thanks for the review. Updated.

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 7]
```diff
@@ -18,34 +18,87 @@ All possible configuration options are available in the
...
+### multus
+
+It is also possible to connect VMI to a secondary networks using
+[Multus](https://github.com/intel/multus-cni). This assumes that multus is
+installed accross your cluster and a corresponding
+`NetworkAttachmentDefinition` CRD was created.
 
-> **Note:** Currently, the only supported network type is `pod`. In the future,
-> more network types may be supported.
+First of all, a `NetworkAttachmentDefinition` for the desired secondary network
```
Comment (rmohr) [inline workloads/virtual-machines/interfaces-and-networks.md]: How about this?

~First of all, a `NetworkAttachmentDefinition` for the desired secondary network
must be created.~ **The** following example defines a network which uses the [ovs-cni
plugin](https://github.com/kubevirt/ovs-cni), which will connect the VMI to Open
vSwitch**'s** bridge `br1` and VLAN 100. Other CNI plugins such as ptp, bridge,
macvlan or flannel might be used as well**.** **For** their installation and usage refer
to **the** respective project documentation.

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 8]
```diff
@@ -18,34 +18,87 @@ All possible configuration options are available in the
...
+must be created. Following example defines a network which uses the [ovs-cni
+plugin](https://github.com/kubevirt/ovs-cni), which will connect the VMI to Open
+vSwitch bridge `br1` and VLAN 100. Other CNI plugins such as ptp, bridge,
+macvlan or flannel might be used as well, for their installation and usage refer
+to respective project documentation.
+
+`NetworkAttachmentDefinition` is usually created by the administrator and just
```
Comment (rmohr) [inline workloads/virtual-machines/interfaces-and-networks.md]: First the `NetworkAttachmentDefinition` needs to be created. That is usually done by an administrator. Users can then reference the definition.

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 9]
```diff
@@ -18,34 +18,87 @@ All possible configuration options are available in the
...
+spec:
+  config: '{
+      "cniVersion": "0.3.1",
+      "type": "ovs",
+      "bridge": "br1",
+      "vlan": 100
+    }'
+```
+
+With following definition, the VMI will be connected the default pod network
```
Comment (rmohr) [inline workloads/virtual-machines/interfaces-and-networks.md]: the VMI will be connected **to** the default pod network

**Comment by phoracek** [comment]
@rmohr thanks, fixed.

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 10]
```diff
@@ -18,34 +18,86 @@ All possible configuration options are available in the
...
+        - name: default
+          bridge: {}
+  networks:
+  - name: default
+    pod: {} # Stock pod network
+```
+
+### multus
+
+It is also possible to connect VMI to a secondary networks using
```
Comment (rmohr) [inline workloads/virtual-machines/interfaces-and-networks.md]: Either `the VMI` or `VMIs`.

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 11]
```diff
@@ -18,34 +18,86 @@ All possible configuration options are available in the
...
 
-> **Note:** Currently, the only supported network type is `pod`.
-
 ## Backend
 
 Network backends are configured in `spec.networks`. A network must have a
 unique name. Additional fields declare which logical or physical device the
 network relates to.
 
+Each network should declare its type by defining on of the following fields:
```
Comment (rmohr) [inline workloads/virtual-machines/interfaces-and-networks.md]: s/on/one/

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 12]
```diff
@@ -18,34 +18,86 @@ All possible configuration options are available in the
...
+        - name: default
+          bridge: {}
+  networks:
+  - name: default
+    pod: {} # Stock pod network
+```
+
+### multus
+
+It is also possible to connect VMIs to a secondary networks using
```
Comment (stu-gott) [inline workloads/virtual-machines/interfaces-and-networks.md]: either `to a secondary network` or `to secondary networks`, but not both.

**Comment by phoracek** [comment]
@stu-gott thanks, fixed

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 13]
```diff
@@ -18,34 +18,86 @@ All possible configuration options are available in the
...
+done by an administrator. Users can then reference the definition.
+
+```yaml
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: ovs-vlan-100
+spec:
+  config: '{
+      "cniVersion": "0.3.1",
```
Comment (myakove) [inline workloads/virtual-machines/interfaces-and-networks.md]: From where do the administrator get cniVersion, how the administrator know this version should be changed?
Or this is some static version that never changed?

Comment (SchSeba) [reply workloads/virtual-machines/interfaces-and-networks.md]: This is something related to every specific cni plugin.
every plugin can support different versions.

**Pull request merged**